tools/pyocd.py

```diff
@@ -0,0 +1,276 @@
+                if args.width == 8:
+                    data = target.readBlockMemoryUnaligned8(args.read, args.len)
+                elif args.width == 16:
+                    if args.read & 0x1:
+                        raise ToolError("read address 0x%08x is not 16-bit aligned" % args.read)
+
+                    byteData = target.readBlockMemoryUnaligned8(args.read, args.len * 2)
+                    i = 0
+                    data = []
+                    while i < len(byteData):
+                        data.append(byteData[i] | (byteData[i+1] << 8))
+                        i += 2
+                elif args.width == 32:
+                    if args.read & 0x3:
+                        raise ToolError("read address 0x%08x is not 32-bit aligned" % args.read)
+
+                    data = target.readBlockMemoryAligned32(args.read, args.len / 4)
+
+                # Either print disasm or hex dump of output
+                if args.disasm:
+                    if args.width == 8:
+                        code = bytearray(data)
+                    elif args.width == 16:
+                        code = bytearray(byteData)
+                    elif args.width == 32:
+                        byteData = []
+                        for v in data:
+                            byteData.extend([(v >> 24) & 0xff, (v >> 16) & 0xff, (v >> 8) & 0xff, v & 0xff])
+                        code = bytearray(byteData)
+
+                    self.disasm(str(code), args.read)
+                else:
+                    dumpHexData(data, args.read, width=args.width)
+            elif args.action == ACTION_WRITE:
+                if args.width == 8:
+                    target.writeBlockMemoryUnaligned8(args.write, args.data)
+                elif args.width == 16:
+                    if args.write & 0x1:
+                        raise ToolError("write address 0x%08x is not 16-bit aligned" % args.write)
+
+                    print "16-bit writes are currently not supported"
+                elif args.width == 32:
+                    if args.write & 0x3:
+                        raise ToolError("write address 0x%08x is not 32-bit aligned" % args.write)
+
+                    target.writeBlockMemoryAligned32(args.write, args.data)
+            elif args.action == ACTION_PROGRAM:
+                if not os.path.exists(args.file):
+                    raise ToolError("%s does not exist!" % args.file)
+
+                print "Programming %s into flash..." % args.file
+                flash.flashBinary(args.file)
+            elif args.action == ACTION_ERASE:
+                if not didErase:
+                    target.massErase()
+            elif args.action == ACTION_UNLOCK:
+                target.unlockFlash()
+            elif args.action == ACTION_GO:
+                target.resume()
+
+        except ToolError, e:
+            print "Error:", e
+            exitCode = 1
+        finally:
+            if board != None:
+                board.uninit()
```

Comment:
By default board.uninit() calls target.resume().  Because of this --halt does not work as expected.  You could fix this by adding changing the line to board.uninit(False)